rest/config.go

```diff
@@ -1284,6 +1285,123 @@ func startServer(config *ServerConfig, sc *ServerContext) {
 	config.Serve(*config.Interface, CreatePublicHandler(sc))
 }
 
+func TestAuth(cluster *gocb.Cluster, username, password string, bucket string, requirePermissions ...string) (statusCode int, err error) {
+	_ = cluster
+	// TODO: Replace with cluster get Management endpoints
+	managementEndpoint := "http://work-couchbase.lan:8091"
+
+	// TODO: Get http client from cluster
+	httpClient := http.DefaultClient
+
+	body := strings.Join(requirePermissions, ",")
+	checkPermissionsURL := fmt.Sprintf("%s/pools/default/checkPermissions", managementEndpoint)
+
+	req, err := http.NewRequest("POST", checkPermissionsURL, bytes.NewBufferString(body))
```

Comment:
Somewhere in this call stack I expect we'll want to add retry for transient HTTP errors.